float64[] coeffX
float64[] coeffY
float64[] coeffZ
float64 offset
